softioc/device.py

```diff
@@ -216,9 +223,14 @@ def _process(self, record):
         if record.PACT:
             return EPICS_OK
 
+        # Ignore memoized value, retrieve it from the VAL field directly later
+        _, severity, alarm = self._value
+
+        self.process_severity(record, severity, alarm)
+
         value = self._read_value(record)
```

Comment:
Put reading `self._value` and `self._read_value(record)` on adjacent lines, then it's much clearer that they're working together and you might not need the comment!